api/envoy/config/api_listener/v2/BUILD

```diff
@@ -0,0 +1,25 @@
+load("@envoy_api//bazel:api_build_system.bzl", "api_proto_library_internal", "api_proto_package")
+
+licenses(["notice"])  # Apache 2
+
+package_group(
+    name = "friends",
+    packages = [
+        "//envoy/api/v2",
+    ],
+)
+
+api_proto_package(
+    deps = [
+        "//envoy/config/filter/network/http_connection_manager/v2:http_connection_manager",
```

Comment:
I think this should be `v2:pkg`.

Reply:
I tried this, but I ran into a circular dependency problem with the go protos: http_connection_manager.proto depends on rds.proto, which is the same directory as lds.proto, so we cannot make lds.proto transitively depend on http_connection_manager.proto.

After chatting with @kyessenov, @htuch, and @dfawley about this, I've changed the `ApiListener` proto to contain an `Any` field instead of a oneof, and I've documented that the type of the `Any` field dictates the type of API listener.  I've also added a large comment saying that in v3, we need to fix the circular dependency and replace the `Any` with a oneof.

I'm not terribly happy about this, but it seems like it's the best we can do in v2.